packages/bpk-component-bottom-sheet/src/BpkBottomSheet.tsx

```diff
@@ -15,86 +15,145 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { useState, type ReactNode } from "react";
+import type { ReactNode } from "react";
+import { useState } from "react";
 
-import { Portal, cssModules } from "../../bpk-react-utils";
-import { withScrim } from "../../bpk-scrim-utils";
+import BpkBreakpoint, { BREAKPOINTS } from "../../bpk-component-breakpoint";
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import { BpkButtonLink } from '../../bpk-component-link';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import BpkCloseButton from '../../bpk-component-close-button';
+import { BpkDialogWrapper, cssModules } from "../../bpk-react-utils";
+import BpkNavigationBar from "../../bpk-component-navigation-bar";
 
 import STYLES from './BpkBottomSheet.module.scss';
-import BpkBottomSheetInner from "./BpkBottomSheetInner";
-
 
 const getClassName = cssModules(STYLES);
-const ScrimBpkBottomSheetInner = withScrim(BpkBottomSheetInner)
 
 export type Props = {
   actionText?: string;
+  ariaLabelledby: string;
   children: ReactNode;
   closeLabel?: string;
   closeOnEscPressed?: boolean;
   closeOnScrimClick?: boolean;
   id: string;
-  isOpen: boolean;
   onAction?: () => void;
   onClose: (
-    arg0?: TouchEvent | MouseEvent | KeyboardEvent,
+    arg0?: Event | KeyboardEvent | MouseEvent | PointerEvent | TouchEvent,
```

Comment:
Does the generic `Event` type not include all the other event types?

Reply:
But doesn't `Event` include all the others, so we could just remove the others then? 🤔
What's the reason we need to call `onClose` with `e.nativeEvent` rather than just the event?